bodo/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -0,0 +1,208 @@
+<#--
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to you under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+-->
+
+JoinType LeftSemiJoin() :
+{
+}
+{
+    <LEFT> <SEMI> <JOIN> { return JoinType.LEFT_SEMI_JOIN; }
+}
+
+SqlNode DateFunctionCall() :
+{
+    final SqlFunctionCategory funcType = SqlFunctionCategory.USER_DEFINED_FUNCTION;
+    final SqlIdentifier qualifiedName;
+    final Span s;
+    final SqlLiteral quantifier;
+    final List<? extends SqlNode> args;
+}
+{
+    <DATE> {
+        s = span();
+        qualifiedName = new SqlIdentifier(unquotedIdentifier(), getPos());
+    }
+    args = FunctionParameterList(ExprContext.ACCEPT_SUB_QUERY) {
+        quantifier = (SqlLiteral) args.get(0);
+        args.remove(0);
+        return createCall(qualifiedName, s.end(this), funcType, quantifier, args);
+    }
+}
+
+SqlNode DateaddFunctionCall() :
+{
+    final SqlFunctionCategory funcType = SqlFunctionCategory.USER_DEFINED_FUNCTION;
+    final Span s;
+    final SqlIdentifier qualifiedName;
+    final TimeUnit unit;
+    final List<SqlNode> args;
+    SqlNode e;
+}
+{
+    ( <DATEADD> | <DATEDIFF> | <DATE_PART> ) {
+        s = span();
+        qualifiedName = new SqlIdentifier(unquotedIdentifier(), getPos());
+    }
+    <LPAREN> unit = TimeUnit() {
+        args = startList(new SqlIntervalQualifier(unit, null, getPos()));
+    }
+    (
+        <COMMA> e = Expression(ExprContext.ACCEPT_SUB_QUERY) {
+            args.add(e);
+        }
+    )*
+    <RPAREN> {
+        return createCall(qualifiedName, s.end(this), funcType, null, args);
+    }
+}
+
+boolean IfNotExistsOpt() :
```

Comment:
It seems to me like a lot of this behavior is copied. I think we should consider limiting this to only those features we intend to support.